sol/compass-cw/src/contract.rs

```diff
@@ -0,0 +1,188 @@
+use crate::msg::{
+    Consensus, CustomResponseMsg, ExecuteMsg, InstantiateMsg, LogicCallSigningData, MigrateMsg,
+    QueryMsg, TargetContractInfo, Valset,
+};
+use crate::msg::{LogicCallArgs, Signature};
+use crate::state::{LAST_CHECKPOINT, MESSAGE_ID_USED, TURNSTONE_ID};
+use crate::{hex, keccak256};
+use cosmwasm_std::{
+    to_binary, Addr, Binary, CosmosMsg, Deps, DepsMut, Env, MessageInfo, Response, StdResult,
+    Uint256,
+};
+use eyre::{bail, ensure, Result};
+use sha3::Digest;
+use std::iter::zip;
+
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use sha3::Keccak256;
+
+/// 2/3 * 2**32.
+/// Validator powers will be normalized to sum to 2**32 in every valset update.
+const POWER_THRESHOLD: u128 = 2_863_311_530;
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn migrate(_deps: DepsMut, _env: Env, _msg: MigrateMsg) -> StdResult<Response> {
+    Ok(Response::new())
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response> {
+    TURNSTONE_ID.save(deps.storage, &msg.turnstone_id)?;
```

Comment:
```suggestion
    SMART_CONTRACT_UNIQUE_ID.save(deps.storage, &msg.smart_contract_unique_id)?;
```
so that it matches what we have in Paloma